src/views/Document.vue

```diff
@@ -118,32 +118,188 @@ export default {
 				.get(`/api/documents/${id}`)
 				.then((response) => response.data);
 		},
+		colorForIndex: function(i, asString = true) {
+			const values = this.HSVToRGB((i * 0.618033988749895) % 1.0, 0.5, Math.sqrt(1.0 - ((i * 0.618033988749895) % 0.5)));
+			return asString ? `rgb(${values[0]}, ${values[1]}, ${values[2]})` : values;
+		},
+		clamp: function(val, min, max) {
+			return Math.max(min, Math.min(val, max));
+		},
+		HSVToRGB: function(h, s, v) {
+			if (s == 0) {
+				return [v, v, v];
+			}
+			if (v == 0) {
+				return [0, 0, 0];
+			}
+			const col = [0, 0, 0];
+			const hval = h * 6;
+			const sel = Math.floor(hval);
+			const mod = hval - sel;
+			const v1 = v * (1 - s);
+			const v2 = v * (1 - s * mod);
+			const v3 = v * (1 - s * (1 - mod));
+
+			switch (sel + 1) {
+				case 0:
+					col[0] = v;
+					col[1] = v1;
+					col[2] = v2;
+					break;
+				case 1:
+					col[0] = v;
+					col[1] = v3;
+					col[2] = v1;
+					break;
+				case 2:
+					col[0] = v2;
+					col[1] = v;
+					col[2] = v1;
+					break;
+				case 3:
+					col[0] = v1;
+					col[1] = v;
+					col[2] = v3;
+					break;
+				case 4:
+					col[0] = v1;
+					col[1] = v2;
+					col[2] = v;
+					break;
+				case 5:
+					col[0] = v3;
+					col[1] = v1;
+					col[2] = v;
+					break;
+				case 6:
+					col[0] = v;
+					col[1] = v1;
+					col[2] = v2;
+					break;
+				case 7:
+					col[0] = v;
+					col[1] = v3;
+					col[2] = v1;
+					break;
+			}
+
+			col[0] = Math.floor(this.clamp(col[0], 0, 1) * 255);
+			col[1] = Math.floor(this.clamp(col[1], 0, 1) * 255);
+			col[2] = Math.floor(this.clamp(col[2], 0, 1) * 255);
+			return col;
+		},
+		mergeIntervals: function (intervals) {
+			
+			if(intervals.length <= 1)
+				return intervals;
+
+			let stack = [];
+			let top = null;
+
+			intervals = intervals.sort(function (a, b) {
+				if (a.from === b.from){
+					return +(a.to > b.to) || +(a.to === b.to) - 1;
+				}
+				return +(a.from > b.from) || - 1;
+			});
+			
+			intervals[0].matches = [{doc: intervals[0].name, per: intervals[0].percentage}]
+			stack.push(intervals[0]);
+			
+			for (var i = 1; i < intervals.length; i++) {
+				top = stack[stack.length - 1];
+				
+				if (top.to < intervals[i].from) {
+					intervals[i].matches = [{doc: intervals[i].name, per: intervals[i].percentage}]
+					stack.push(intervals[i]);
+				}
+				
+				else if (top.to <= intervals[i].to){
+					top.to = intervals[i].to;
+					top.matches.push({doc: intervals[i].name, per: intervals[i].percentage})
+					stack.pop();
+					stack.push(top);
+				}
+				else if(top.to > intervals[i].to){
+					top.matches.push({doc: intervals[i].name, per: intervals[i].percentage})
+					stack.pop();
+					stack.push(top);
+				}
+			}
+			
+			return stack;
+		},
+		toggleTooltip: function(docs){
+
+			const docs_unique = docs.filter((elem, index) => docs.findIndex(obj => obj.doc === elem.doc) === index)
```

Comment:
Set doesn't work, only Map. 

![image](https://user-images.githubusercontent.com/32281312/111210232-46599d00-85cd-11eb-878d-491cd575381d.png)


I've tried indexOf before but also doesn't work, because I need to filter document name and also percentage, this filters only document name.